### PR TITLE
JAMES-3444 Allow moving JMAP mailets in a local-delivery processor

### PR DESCRIPTION
**Pull request by chibenwa**
Reduces LDAP calls

### CODE REVIEW

**Review by rouazana — COMMENTED** [review]
nice but "transport" and "local-delivery" seems to be some magic values

**Comment by chibenwa** [comment]
> nice but "transport" and "local-delivery" seems to be some magic values

You know my position: kill these checks, that can be easily cheated, and brings limited safety.

I'd be happy we reach a consensus to do so.

In the meantime, I think we are stuck with "magic values".

**Comment by chibenwa** [comment]
At leats now it is documented in the Mail interface...

**Comment by rouazana** [comment]
> At leats now it is documented in the Mail interface...

Thanks, that's a little better.

**Review by Arsnael — APPROVED** [review]

File: server/container/guice/protocols/jmap/src/test/java/org/apache/james/jmap/draft/MailetPreconditionTest.java [hunk 1]
```diff
@@ -61,25 +62,33 @@ void vacationMailetCheckShouldThrowOnNullList() {
...
                 .isInstanceOf(ConfigurationException.class);
         }
 
         @Test
         void vacationMailetCheckShouldNotThrowIfValidPairPresent() {
             List<MatcherMailetPair> pairs = Lists.newArrayList(new MatcherMailetPair(new RecipientIsLocal(), new VacationMailet(null, null, null, null, null)));
 
-            assertThatCode(() -> JMAPModule.VACATION_MAILET_CHECK.check(pairs))
+            assertThatCode(() -> JMAPModule.VACATION_MAILET_CHECK.check(ImmutableMultimap.of(
+                "transport", new MatcherMailetPair(new RecipientIsLocal(), new VacationMailet(null, null, null, null, null)))))
```
Comment (LanKhuat) [inline server/container/guice/protocols/jmap/src/test/java/org/apache/james/jmap/draft/MailetPreconditionTest.java:82]: dead code above?

File: server/container/guice/protocols/jmap/src/test/java/org/apache/james/jmap/draft/MailetPreconditionTest.java [hunk 2]
```diff
@@ -61,25 +62,33 @@ void vacationMailetCheckShouldThrowOnNullList() {
...
+                .doesNotThrowAnyException();
+        }
+
+        @Test
+        void vacationMailetCheckShouldSupportLocalDeliveryProcessor() {
+            List<MatcherMailetPair> pairs = Lists.newArrayList(new MatcherMailetPair(new RecipientIsLocal(), new VacationMailet(null, null, null, null, null)));
```
Comment (LanKhuat) [inline server/container/guice/protocols/jmap/src/test/java/org/apache/james/jmap/draft/MailetPreconditionTest.java:88]: dead code?

**Review by LanKhuat — APPROVED** [review]

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**